Generation/src/PythiaInterface.cpp

```diff
@@ -53,55 +66,89 @@ StatusCode PythiaInterface::execute() {
+  m_iAbort=0;
+
+  // Print debug: Pythia event info
+  if (msgLevel() <= MSG::DEBUG) {
+
+    for (int i = 0; i < m_pythia->event.size(); ++i){
+
       //if (m_pythia->event[i].isFinal() && m_pythia->event[i].isCharged())
-      std::cout << "PythiaInterface : id : stat : px : py : pz : e : m : " 
-                << " " << m_pythia->event[i].id()
-                << " " << m_pythia->event[i].status()
-                //<< " " << m_pythia->event[i].mother1()
-                //<< " " << m_pythia->event[i].mother2()
-                //<< " " << m_pythia->event[i].daughter1()
-                //<< " " << m_pythia->event[i].daughter2()
-                << " " << m_pythia->event[i].px()
-                << " " << m_pythia->event[i].py()
-                << " " << m_pythia->event[i].pz() 
-                << " " << m_pythia->event[i].e()
-                << " " << m_pythia->event[i].m()
+      std::cout << "PythiaInterface : "
+                << " Id: "       << setw(3) << i
+                << " PDG: "      << setw(5) << m_pythia->event[i].id()
+                << " Mothers: "  << setw(3) << m_pythia->event[i].mother1()   << " -> " << setw(3) << m_pythia->event[i].mother2()
+                << " Daughters:" << setw(3) << m_pythia->event[i].daughter1() << " -> " << setw(3) << m_pythia->event[i].daughter2()
+                << " Stat: "     << setw(2) << m_pythia->event[i].status()
+                << setprecision(2) << " Px: " << setw(9) << m_pythia->event[i].px()
+                << setprecision(2) << " Py: " << setw(9) << m_pythia->event[i].py()
+                << setprecision(2) << " Pz: " << setw(9) << m_pythia->event[i].pz()
+                << setprecision(2) << " E: "  << setw(9) << m_pythia->event[i].e()
+                << setprecision(2) << " M: "  << setw(9) << m_pythia->event[i].m()
                 << std::endl;
-  }
-  */
+    }
+  } // Debug
 
-  /// Construct new emhepmcevtpty HepMC event
+  // Define HepMC event and convert Pythia event into this HepMC event type
   HepMC::GenEvent* theEvent = new HepMC::GenEvent( HepMC::Units::GEV, HepMC::Units::MM);
-  toHepMC->fill_next_event(*m_pythia,theEvent);
+  toHepMC->fill_next_event(*m_pythia, theEvent, m_iEvent);
   //theEvent-> print();
-  /*  
-
-  for (HepMC::GenEvent::particle_iterator ipart = theEvent->particles_begin() ;
-       ipart!=theEvent->particles_end(); ++ipart)
-       std::cout << "HepMC : id : stat : px : py : pz : e : m : " 
-              << (*ipart)->pdg_id()
-              << " " << (*ipart)->status()
-              << " " << (*ipart)->momentum().px()
-              << " " << (*ipart)->momentum().py()
-              << " " << (*ipart)->momentum().pz() 
-              << " " << (*ipart)->momentum().e()
-              << " " << (*ipart)->momentum().m()
-              << std::endl;
-  */
 
+  // Print debug: HepMC event info
+  if (msgLevel() <= MSG::DEBUG) {
+
+    for (HepMC::GenEvent::particle_iterator ipart = theEvent->particles_begin() ;
```

Comment:
not very important as it is in the debugging clause, but this would be nicer:
```c++
for (auto ipart = theEvent->particles_begin(), end = theEvent->particles_end();
      ipart != end; ++ipart) {
```